### PR TITLE
Listview: Implement classes option

### PR DESCRIPTION
**Pull request by gabrielschulhof**

### CODE REVIEW

File: js/widgets/listview.js [hunk 1]
```diff
@@ -106,48 +92,43 @@ return $.widget( "mobile.listview", $.extend( {
...
+		li = this._getChildrenByTagName( list[ 0 ], "li", "LI" );
 
 		for ( pos = 0, numli = li.length; pos < numli; pos++ ) {
 			item = li.eq( pos );
-			itemClass = "";
+			itemClass = "ui-listview-item";
+			itemExtraClass = undefined;
 
-			if ( create || item[ 0 ].className.search( /\bui-li-static\b|\bui-li-divider\b/ ) < 0 ) {
+			if ( create || item[ 0 ].className
```
Comment (arschmitz) [inline js/widgets/listview.js]: is this actually faster then `hasClass` ?

File: js/widgets/listview.js [hunk 2]
```diff
@@ -106,48 +92,43 @@ return $.widget( "mobile.listview", $.extend( {
...
+					.search( /\bui-listview-item-static\b|\bui-listview-item-divider\b/ ) < 0 ) {
```
Comment (arschmitz) [inline js/widgets/listview.js]: switch to `=== -1`

File: js/widgets/listview.js [hunk 3]
```diff
@@ -211,24 +212,57 @@ return $.widget( "mobile.listview", $.extend( {
...
+			this._addClass.apply( this,
+				[ $( itemClassDict[ dictionaryKey ] ) ]
+					.concat( dictionaryKey.split( "|" ) ) );
 		}
 
-		// Deprecated in 1.4. From 1.5 you have to add class ui-li-has-thumb or ui-li-has-icon to the LI.
-		this._addThumbClasses( li );
-		this._addThumbClasses( li.find( ".ui-button" ) );
+		this._addClass(
+			li.filter( function trueIfContextHasBubbleOrHasAnchorThatHasBubble() {
```
Comment (arschmitz) [inline js/widgets/listview.js]: Would it make sense to extract this out into its own method for code clarity here?

File: tests/integration/listview/index.html [hunk 4]
```diff
@@ -11,18 +11,14 @@
...
-	});
-	</script>
+	<script src="../../../external/qunit-assert-classes/qunit-assert-classes.js"></script>
 	<script>
 		$.testHelper.asyncLoad([
 			[ "widgets/page" ],
 			[
 				"navigation",
 				"widgets/listview",
+				"widgets/listview.backcompat",
```
Comment (arschmitz) [inline tests/integration/listview/index.html]: backcompat tests should be seperate

File: tests/integration/listview/index.html [hunk 5]
```diff
@@ -43,69 +39,47 @@
 	<div id="qunit"></div>
 
 	<!-- Basic Linked view test -->
-	<div data-nstest-role="page" id='basic-linked-test'>
-		<div data-nstest-role="header">
-			<h1>Basic List View</h1>
-		</div>
-		<div data-nstest-role="content">
-			<ul data-nstest-role="listview">
+	<div data-nstest-role="page" id="main-page">
```
Comment (arschmitz) [inline tests/integration/listview/index.html]: non backcompat should replace `nstest` with `ui`

File: tests/integration/listview/index.html [hunk 6]
```diff
@@ -11,18 +11,14 @@
 	<script src="../../jquery.setNameSpace.js"></script>
```
Comment (arschmitz) [inline tests/integration/listview/index.html]: in non backcompat tests this should be removed

**Comment by arschmitz** [comment]
@gabrielschulhof looking good only a few things on first pass going to do some testing on some devices

File: js/widgets/listview.js (hunk 1 above)
Comment (gabrielschulhof) [inline js/widgets/listview.js]: I'd love to find out - unfortunately jsperf is offline again :(

File: js/widgets/listview.js (hunk 3 above)
Comment (gabrielschulhof) [inline js/widgets/listview.js]: Yeah, I suppose I could add it to the top of the IIFE.

File: js/widgets/listview.js [hunk 7]
```diff
@@ -156,35 +137,51 @@ return $.widget( "mobile.listview", $.extend( {
...
+						}
 
 						// Reduce to the first anchor, because only the first gets the buttonClass
 						a = a.first();
 					} else if ( icon ) {
-						buttonClass += " ui-icon-end ui-icon-" + icon;
+						span = createEnhanced ? a.children( ".ui-listview-item-icon" ) :
+							$( "<span>" );
+						this._addClass( span, "ui-listview-item-icon", "ui-icon ui-icon-" + icon +
+							" ui-widget-icon-floatend" );
```
Comment (jaspermdegroot) [inline js/widgets/listview.js]: ``` ui-widget-icon-floatend``` --> ``` ui-widget-icon-float``` (see https://github.com/jquery/jquery-mobile/issues/8145#issuecomment-124193627)

**Comment by arschmitz** [comment]
@gabrielschulhof this looks like it has some conflicts now

**Comment by arschmitz** [comment]
@gabrielschulhof just going to run this through the devices again but looks good

**Comment by gabrielschulhof** [comment]
@arschmitz I'll rebase again so it doesn't have conflicts.

**Comment by arschmitz** [comment]
:+1:

**Pull request closed without merging**